### PR TITLE
Brid player component

### PR DESCRIPTION
**Pull request by DarXector**
This is the amp-brid-player component from the intent to implement issue: #2378

Thank you,
Marko.

### CODE REVIEW

**Comment by cramforce** [comment]
Please fix the lint errors

File: extensions/amp-brid-player/0.1/amp-brid-player.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const width = this.element.getAttribute('width');
```
Comment (cramforce) [inline extensions/amp-brid-player/0.1/amp-brid-player.js]: Any chance you have a poster image you could load at the same time as the iframe?

File: extensions/amp-brid-player/0.1/amp-brid-player.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    if(this.element.getAttribute('data-video')){
+      feedType = "video";
+    } else if(this.element.getAttribute('data-playlist')){
+      feedType = "playlist";
+    }
+
+    //Create iframe
+    const iframe = document.createElement('iframe');
+    let src = "https://services.brid.tv/services/iframe/"+feedType+"/"+feedID+"/"+partnerID+"/"+playerID+"/0/1";
```
Comment (cramforce) [inline extensions/amp-brid-player/0.1/amp-brid-player.js]: Add `encodeURIComponent` around the stuff that is being interpolated into the URL

**Comment by DarXector** [comment]
I have fixed lint errors and added the encodeURIComponent when generating the URL.

As for the poster image, I can generate a poster image URL and add the image to the element, but only for those partners that use our encoding and hosting services. For the partners that use external video and poster URLs, the generated one will 404. We could load the video feed and get the URL from it, this, I guess, goes against the AMP principle?

Also tests always fails here:
```
example brid-player.amp.html should validate FAILED
	AssertionError: expected '/home/travis/build/DarXector/amphtml/examples/brid-player.amp.html:9:4 The attribute \'custom-element\' in tag \'amp-access extension .js script\' is set to the invalid value \'amp-brid-player\'.
```
Should I add something to the validator or is that something that needs to be updated on your side?

Thak you,
Marko.

**Comment by cramforce** [comment]
For the test failure: The test allows to whitelist the failure. We will then update the validator and make it no longer need the whitelist entry.

For the image: maybe you could create a new endpoint that always redirects to a working image? It is definitely a much better experience to have a poster image because it can be drawn in prerender as opposed to the iframe which won't be.

**Comment by DarXector** [comment]
Added poster image support

**Comment by DarXector** [comment]
Anything else?

File: extensions/amp-brid-player/0.1/amp-brid-player.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const playerID = AMP.assert(this.element.getAttribute('data-player'),
+        'The data-player attribute is required for <amp-brid-player> %s',
+        this.element);
+
+    const partnerID = AMP.assert(
```
Comment (dvoytenko) [inline extensions/amp-brid-player/0.1/amp-brid-player.js]: This is already asserted above, no?

**Comment by dvoytenko** [comment]
@DarXector Looks good. Just a small comment and please squash the commits. Otherwise, this looks to be ready to be merged.

One question - have you already filed the request for validator update?

File: extensions/amp-brid-player/0.1/amp-brid-player.js (hunk 3 above)
Comment (DarXector) [inline extensions/amp-brid-player/0.1/amp-brid-player.js]: Yes, but for one of my tests ("requires data-partner" in test-amp-brid-player.js) fails with only the assert located in `buildCallback`. I looks like `buildCallback` runs before `.should.eventually.be.rejectedWith( /The data-partner attribute is required for/);` is added.

Correct me if I'm wrong. 
Or do I need a delay somewhere?

**Comment by DarXector** [comment]
@dvoytenko no. I'll get right to it.

Thank you.

File: extensions/amp-brid-player/0.1/amp-brid-player.js (hunk 3 above)
Comment (dvoytenko) [inline extensions/amp-brid-player/0.1/amp-brid-player.js]: I see. No problem then - tests sometimes have slightly different requirements. We will be addressing that soon.

**Comment by dvoytenko** [comment]
Thanks @DarXector for filing the validator request. Now we just need to squash commits in this PR and ready to go.

**Comment by DarXector** [comment]
Commits squashed

**Comment by dvoytenko** [comment]
LGTM and merging.

**Comment by DarXector** [comment]
Thank you!

**Comment by DarXector** [comment]
When is this going to be available for our clients to use?

Best,
Marko

**Comment by dvoytenko** [comment]
@DarXector According to the https://github.com/ampproject/amphtml/releases it's already in the opt-in canary and can be experimented with. It should be on PROD next Thursday and you can monitor it via the same releases page. But to be used, the #2580 should be also done so that validators whitelist this. I'll ping on that issue to see the ETA.

**Pull request merged**